gossipsub.go

```diff
@@ -522,6 +529,45 @@ func (gs *GossipSubRouter) Attach(p *PubSub) {
 	}
 }
 
+func (gs *GossipSubRouter) manageAddrBook() error {
+	sub, err := gs.p.host.EventBus().Subscribe([]interface{}{
+		&event.EvtPeerIdentificationCompleted{},
+		&event.EvtPeerConnectednessChanged{},
+	})
+	if err != nil {
+		log.Errorf("failed to subscribe to peer identification events: %v", err)
+	}
+	defer sub.Close()
+
+	for {
+		select {
+		case <-gs.p.ctx.Done():
+			return nil
+		case ev := <-sub.Out():
+			switch ev := ev.(type) {
+			case event.EvtPeerIdentificationCompleted:
+				if ev.SignedPeerRecord != nil {
+					cab, ok := peerstore.GetCertifiedAddrBook(gs.cab)
+					if ok {
+						ttl := peerstore.RecentlyConnectedAddrTTL
+						if gs.p.host.Network().Connectedness(ev.Peer) == network.Connected {
```

Comment:
This API will also be confusing for users :(

Reply:
I think for all events we need to do `Network.Connectedness(peer)` checks

Reply:
Maybe this event should have a Connectedness method?

Reply:
Open question (for me): Can we be sure that a ConnectednessChange == Connected event will always happen before a EvtPeerIdentificationCompleted event?

Reply:
(yes)

Reply:
I think we can improve this API, but that shouldn't block this fix. Let's do this for now.